src/charm.py

```diff
@@ -194,8 +198,8 @@ def _get_env_vars(self):
             ],
             "EXECUTOR_SERVER_METRICS_PORT_NAME": config["executor-server-metrics-port-name"],
             "EXECUTOR_SERVER_PORT": config["executor-server-port"],
-            "ISTIO_ENABLED": str(bool(self.model.relations["istio"])).lower(),
-            "ISTIO_GATEWAY": config["istio-gateway"],
+            "ISTIO_ENABLED": str(bool(self.model.relations["gateway-info"])).lower(),
```

Comment:
There is an asymmetry between this and the next line.  `ISTIO_ENABLED` is set by reaching into the relation data right now and inferring something, whereas `ISTIO_GATEWAY` is set by using the data previously retrieved and cached in this class.  They're not necessarily in sync (although it would take a very unlucky situation to actually get different data).

I think it would feel more consistent if we set `ISTIO_ENABLED` from data we've already cached, similar to how `ISTIO_GATEWAY` is set.  And maybe hide any logic in a helper. 
 What do you think?

Reply:
Yes, this might actually create a bug when relations are deleted. Looking into your next comment, I think, I over-engineered the whole thing. 
There should be single new `_get_gateway_info()` that retrieves data (acts as `_process_gateway_info()`) and returns it (acts as current `_get_gateway_info()`), This way we always get latest.
Thoughts?